metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -255,6 +256,44 @@ def get_tableau_auth(
             )
         return authentication
 
+    def check_user_role(
+        self, logged_in_user: LoggedInUser
+    ) -> Dict[Union[SourceCapability, str], CapabilityReport]:
+        capability_dict: Dict[Union[SourceCapability, str], CapabilityReport] = {
+            c.SITE_PERMISSION: CapabilityReport(
+                capable=True,
+            )
+        }
+
+        failure_reason: str = (
+            "The user does not possess the `Site Administrator Explorer` role."
+        )
+
+        mitigation_message: str = "Assign `Site Administrator Explorer` role to the user {}. Refer setup guide: https://datahubproject.io/docs/quick-ingestion-guides/tableau/setup"
+
+        try:
+            # TODO: Add check for `Enable Derived Permissions`
+            if not logged_in_user.is_site_administrator_explorer():
+                capability_dict[c.SITE_PERMISSION] = CapabilityReport(
+                    capable=False,
+                    failure_reason=f"{failure_reason} User current role is {logged_in_user.site_role()}",
+                    mitigation_message=mitigation_message.format(
+                        logged_in_user.user_name()
+                    ),
+                )
+
+            return capability_dict
+
+        except Exception as e:
+            logger.warning(e)
+            capability_dict[c.SITE_PERMISSION] = CapabilityReport(
+                capable=False,
+                failure_reason=failure_reason,
+                mitigation_message=mitigation_message.format(""),  # user is unknown
```

Comment:
won't this leave an extra space before the period?

Reply:
the error message should just say that we were unable to check the user's role information - otherwise it is misleading

Reply:
done